snow/engine/snowman/transitive.go

```diff
@@ -1103,7 +1114,59 @@ func (t *Transitive) addUnverifiedBlockToConsensus(
 	)
 	return true, t.Consensus.Add(ctx, &memoryBlock{
 		Block:   blk,
-		metrics: &t.metrics,
+		metrics: t.metrics,
 		tree:    t.nonVerifieds,
 	})
 }
+
+// getProcessingAncestor finds [initialVote]'s most recent ancestor that is
+// processing in consensus. If no ancestor could be found, false is returned.
+//
+// Note: If [initialVote] is processing, then [initialVote] will be returned.
+func (t *Transitive) getProcessingAncestor(ctx context.Context, initialVote ids.ID) (ids.ID, bool) {
```

Comment:
This was moved into this file because all of the fields that are used on are this struct. I was considering moving this function off the structs entirely and just passing in the fields... But there are quite a few things that can be accessed... Especially because of the `getBlock` call.